From ba2fa7495be20b447fd189f36ceb4df82093c527 Mon Sep 17 00:00:00 2001
From: Siddharth Seth <sseth@apache.org>
Date: Mon, 3 Oct 2016 11:06:07 -0700
Subject: [PATCH 688/723] CDH-46097: HIVE-14810:. Fix failing test:
 TestMetaStoreMetrics.testMetaDataCounts. (Barna
 Zsombor Klara, reviewed by Siddharth Seth)

(cherry picked from commit ee0465b2550c0fcf05a11b6ecaeb17502b534e14)

Change-Id: Ie799ccdf4649c7b891c4e39bb87d988cb40502a3
---
 .../hive/metastore/TestMetaStoreMetrics.java       |   21 +++++++++++++-------
 1 file changed, 14 insertions(+), 7 deletions(-)

diff --git a/itests/hive-unit/src/test/java/org/apache/hadoop/hive/metastore/TestMetaStoreMetrics.java b/itests/hive-unit/src/test/java/org/apache/hadoop/hive/metastore/TestMetaStoreMetrics.java
index 3f08b88..98e2f03 100644
--- a/itests/hive-unit/src/test/java/org/apache/hadoop/hive/metastore/TestMetaStoreMetrics.java
+++ b/itests/hive-unit/src/test/java/org/apache/hadoop/hive/metastore/TestMetaStoreMetrics.java
@@ -74,6 +74,16 @@ public void testMethodCounts() throws Exception {
 
   @Test
   public void testMetaDataCounts() throws Exception {
+    CodahaleMetrics metrics = (CodahaleMetrics) MetricsFactory.getInstance();
+    String json = metrics.dumpJson();
+
+    int initDbCount = (new Integer((MetricsTestUtils.getJsonNode(json, MetricsTestUtils.GAUGE,
+        MetricsConstant.INIT_TOTAL_DATABASES)).asText())).intValue();
+    int initTblCount = (new Integer((MetricsTestUtils.getJsonNode(json, MetricsTestUtils.GAUGE,
+        MetricsConstant.INIT_TOTAL_TABLES)).asText())).intValue();
+    int initPartCount = (new Integer((MetricsTestUtils.getJsonNode(json, MetricsTestUtils.GAUGE,
+        MetricsConstant.INIT_TOTAL_PARTITIONS)).asText())).intValue();
+
     //1 databases created
     driver.run("create database testdb1");
 
@@ -110,9 +120,7 @@ public void testMetaDataCounts() throws Exception {
     driver.run("use default");
     driver.run("drop database tempdb cascade");
 
-    //give timer thread a chance to print the metrics
-    CodahaleMetrics metrics = (CodahaleMetrics) MetricsFactory.getInstance();
-    String json = metrics.dumpJson();
+    json = metrics.dumpJson();
     MetricsTestUtils.verifyMetricsJson(json, MetricsTestUtils.COUNTER, MetricsConstant.CREATE_TOTAL_DATABASES, 2);
     MetricsTestUtils.verifyMetricsJson(json, MetricsTestUtils.COUNTER, MetricsConstant.CREATE_TOTAL_TABLES, 7);
     MetricsTestUtils.verifyMetricsJson(json, MetricsTestUtils.COUNTER, MetricsConstant.CREATE_TOTAL_PARTITIONS, 9);
@@ -127,11 +135,10 @@ public void testMetaDataCounts() throws Exception {
     baseHandler.init();
     baseHandler.updateMetrics();
 
-    //1 new db + default
     json = metrics.dumpJson();
-    MetricsTestUtils.verifyMetricsJson(json, MetricsTestUtils.GAUGE, MetricsConstant.INIT_TOTAL_DATABASES, 2);
-    MetricsTestUtils.verifyMetricsJson(json, MetricsTestUtils.GAUGE, MetricsConstant.INIT_TOTAL_TABLES, 4);
-    MetricsTestUtils.verifyMetricsJson(json, MetricsTestUtils.GAUGE, MetricsConstant.INIT_TOTAL_PARTITIONS, 6);
+    MetricsTestUtils.verifyMetricsJson(json, MetricsTestUtils.GAUGE, MetricsConstant.INIT_TOTAL_DATABASES, initDbCount + 1);
+    MetricsTestUtils.verifyMetricsJson(json, MetricsTestUtils.GAUGE, MetricsConstant.INIT_TOTAL_TABLES, initTblCount + 4);
+    MetricsTestUtils.verifyMetricsJson(json, MetricsTestUtils.GAUGE, MetricsConstant.INIT_TOTAL_PARTITIONS, initPartCount + 6);
   }
 
 
-- 
1.7.9.5

